src/main/java/com/avispl/dal/communicator/polycom/videoos/PolycomVideoOS.java

```diff
@@ -0,0 +1,935 @@
+        } catch (CommandFailureException cfe) {
+            if (cfe.getStatusCode() == 404) {
+                if(logger.isDebugEnabled()) {
+                    logger.debug("Conference " + conferenceId + " is not available anymore. Skipping media stats retrieval.");
+                }
+                return JsonNodeFactory.instance.arrayNode();
+            } else {
+                logger.error("Unable to find conference by id " + conferenceId);
+                throw cfe;
+            }
+        }
+    }
+
+    /**
+     * Request device reboot
+     *
+     * @throws Exception during http communication
+     */
+    private boolean reboot() throws Exception {
+        ObjectNode rebootRequest = JsonNodeFactory.instance.objectNode();
+        rebootRequest.put("action", "reboot");
+        boolean success = doPost(REBOOT, rebootRequest).getStatusCode().is2xxSuccessful();
+        if (success) {
+            sessionId = null;
+            internalDestroy();
+        }
+        return success;
+    }
+
+    /**
+     * Retrieve device system information, including hardware info and lan status
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    public void getSystemInfo(Map<String, String> statistics) throws Exception {
+        JsonNode response = doGet(buildHttpUrl(SYSTEM), JsonNode.class);
+        if (response == null) {
+            return;
+        }
+        addStatisticsProperty(statistics, "System#Serial Number", response.get("serialNumber"));
+        addStatisticsProperty(statistics, "System#Software Version", response.get("softwareVersion"));
+        addStatisticsProperty(statistics, "System#System State", response.get("state"));
+        addStatisticsProperty(statistics, "System#System Name", response.get("systemName"));
+        addStatisticsProperty(statistics, "System#System Uptime", response.get("uptime"));
+        addStatisticsProperty(statistics, "System#System Build", response.get("build"));
+        addStatisticsProperty(statistics, "System#System Reboot Needed", response.get("rebootNeeded"));
+        addStatisticsProperty(statistics, "System#Device Model", response.get("model"));
+        addStatisticsProperty(statistics, "System#Device Hardware Version", response.get("hardwareVersion"));
+
+        if (response.get("lanStatus") == null) {
+            return;
+        }
+        addStatisticsProperty(statistics, "Lan Status#Duplex", response.get("lanStatus").get("duplex"));
+        addStatisticsProperty(statistics, "Lan Status#Speed Mbps", response.get("lanStatus").get("speedMbps"));
+        addStatisticsProperty(statistics, "Lan Status#State", response.get("lanStatus").get("state"));
+    }
+
+    /**
+     * Retrieve list of device's applications
+     *
+     * @param statistics map to put values to
+     * @throws Exception during http communication
+     */
+    public void getApplications(Map<String, String> statistics) throws Exception {

```

Comment:
Similar to above... Is there a reason this method is public? also, name should start with "retrieve" instead of "get"